src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -641,6 +648,61 @@ def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None
         # super().post_remove(daemon)
         pass
 
+    def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
+        daemon_spec.final_config, daemon_spec.deps = super().generate_config(daemon_spec)
+
+        mon_spec = cast(MONSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
```

Comment:
This is causing upgrade tests in teuthology to fail
```
[WRN] UPGRADE_REDEPLOY_DAEMON: Upgrading daemon mon.b on host smithi047 failed.
    Upgrade daemon: mon.b: Service mon not found.
```
We're assuming there will always be a mon spec here. In a normal deployment there always would be, but teuthology is special and there might not. Will need to take this into account and check for it here.
https://pulpito.ceph.com/adking-2023-05-01_12:44:48-orch:cephadm-wip-adk3-testing-2023-04-25-1440-quincy-distro-default-smithi/7259569

Reply:
> This is causing upgrade tests in teuthology to fail
> 
> ```
> [WRN] UPGRADE_REDEPLOY_DAEMON: Upgrading daemon mon.b on host smithi047 failed.
>     Upgrade daemon: mon.b: Service mon not found.
> ```
> 
> We're assuming there will always be a mon spec here. In a normal deployment there always would be, but teuthology is special and there might not. Will need to take this into account and check for it here. https://pulpito.ceph.com/adking-2023-05-01_12:44:48-orch:cephadm-wip-adk3-testing-2023-04-25-1440-quincy-distro-default-smithi/7259569

added https://github.com/ceph/ceph/pull/51321 to this